packages/contracts/token_contract/tests/utils/mod.rs

```diff
@@ -0,0 +1,186 @@
+        wallet: WalletUnlocked,
+        amount: u64,
+    ) -> FuelCallResponse<u64> {
+        let tx_params = TxParameters::new(None, Some(1_000_000), None);
+        let call_params = CallParameters::new(Some(amount), Some(BASE_ASSET_ID), Some(1_000_000));
+
+        contract
+            .with_wallet(wallet)
+            .unwrap()
+            .methods()
+            .deposit()
+            .tx_params(tx_params)
+            .call_params(call_params)
+            .expect("REASON")
+            .call()
+            .await
+            .unwrap()
+    }
+
+    pub async fn mint_tokens(
+        contract: &TokenContract,
+        asset_id: ContractId,
+        wallet: WalletUnlocked,
+    ) -> FuelCallResponse<u64> {
+        contract
+            .with_wallet(wallet)
+            .unwrap()
+            .methods()
+            .mint(asset_id)
+            .append_variable_outputs(2)
+            .call()
+            .await
+            .unwrap()
+    }
+}
+
+pub mod test_helpers {
+    use super::*;
+
+    pub async fn setup(initial_params: Option<SetupParams>) -> Metadata {
+        const ASSET_CONTRACT_BINARY_PATH: &str = "./out/debug/token_contract.bin";
+        const ASSET_CONTRACT_STORAGE_PATH: &str = "./out/debug/token_contract-storage_slots.json";
+
+        let params = initial_params.unwrap_or({
+            SetupParams {
+                num_wallets: Some(2),
+                coins_per_wallet: Some(1),
+                coin_amount: Some(1_000_000),
+                mint_price: Some(20),
+                mint_limit: Some(100),
+                total_supply: Some(100),
+            }
+        });
+
+        let num_wallets = params.num_wallets;
+        let coins_per_wallet = params.coins_per_wallet;
+        let coin_amount = params.coin_amount;
+        let mut wallets = launch_custom_provider_and_get_wallets(
+            WalletsConfig::new(num_wallets, coins_per_wallet, coin_amount),
+            None,
+            None,
+        )
+        .await;
+
+        let wallet1 = wallets.pop().unwrap();
+        let wallet2 = wallets.pop().unwrap();
+
+        let contract_id = Contract::deploy(
+            ASSET_CONTRACT_BINARY_PATH,
+            &wallet1,
+            TxParameters::default(),
+            StorageConfiguration::with_storage_path(Some(ASSET_CONTRACT_STORAGE_PATH.to_string())),
+        )
+        .await
+        .unwrap();
+
+        let asset_id = ContractId::new(*contract_id.hash());
+        let token_contract = TokenContract::new(contract_id.clone(), wallet1.clone());
+        let deployer = wallet1;
+        let user = wallet2;
+        let constructor_params = ConstructorParams {
+            mint_price: params.mint_price.unwrap_or_default(),
+            mint_limit: params.mint_limit.unwrap_or_default(),
+            total_supply: params.total_supply.unwrap_or_default(),
+        };
+
+        let provider = deployer.get_provider().unwrap().clone();
+        let metadata = {
+            Metadata {
```

Comment:
I would consider what type of groupings these variables can have e.g. `contracts`, `users` etc. and then return those individual structures. This lets you put a bound on the structure rather than having one big `meta` that is difficult to name